core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -385,6 +385,8 @@ impl SchedulerController {
                 if result.is_err() {
                     saturating_add_assign!(num_dropped_on_age_and_status, 1);
                     self.container.remove_by_id(&id.id);
```

Comment:
Why do we need to remove by ID if we haven't pushed it into the queue? Is this just in case it was already present on the queue?

Reply:
`TransacionStateContainer` has 2 components:
1. a priority queue of `(priority, id)`
2. a map from `id` -> `TransactionState`

We pop from the queue.
If invalid, we remove the `TransactionState` from the map.
If valid, we **should** push it back into the queue so we can process it later when `schedule` pops it. It is still in the map though.

Reply:
```
pub(crate) fn remove_by_id(&mut self, id: &TransactionId) {
        self.id_to_transaction_state
            .remove(id)
            .expect("transaction must exist");
    }
```